src/tests/system/tests/test_sssctl_analyze.py

```diff
@@ -0,0 +1,289 @@
+
+    client.ssh("user1", "Secret123").connect()
+
+    result = client.sssctl.analyze_request("show 1 --pam")
+    assert result.rc == 0
+    assert "CID #1" in result.stdout
+
+    assert "SSS_PAM_AUTHENTICATE" in result.stdout
+    assert "SSS_PAM_ACCT_MGMT" in result.stdout
+    assert "SSS_PAM_SETCRED" in result.stdout
+
+
+@pytest.mark.tools
+@pytest.mark.ticket(bz=2013259)
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_sssctl_analyze__tevent_id(client: Client, provider: GenericProvider):
+    """
+    :title: "sssctl analyze" to parse tevent chain IDs from logs
+    :setup:
+        1. Add user
+        2. Enable debug_level to 9 in the 'nss', 'pam' and domain section
+        3. Start SSSD
+        4. Log in as user via ssh
+    :steps:
+        1. Call sssctl analyze request show 1 --pam
+        2. Confirm tevent chain IDs(RID) is showing in logs
+    :expectedresults:
+        1. Called successfully
+        2. Output is correct
+    :customerscenario: True
+    """
+    provider.user("user1").add()
+    client.sssd.nss["debug_level"] = "9"
+    client.sssd.pam["debug_level"] = "9"
+    client.sssd.domain["debug_level"] = "9"
+    client.sssd.start()
+
+    client.ssh("user1", "Secret123").connect()
+
+    result = client.sssctl.analyze_request("show 1 --pam")
+    assert result.rc == 0
+    assert "RID#" in result.stdout, "RID# was not found in the output"
+    assert "user1@test" in result.stdout, "user1@test was not found in the output"
+
+
+@pytest.mark.tools
+@pytest.mark.ticket(bz=2013260)
+@pytest.mark.topology(KnownTopologyGroup.AnyAD)
+def test_sssctl_analyze__parse_child_logs(client: Client, provider: GenericADProvider):
+    """
+    :title: "sssctl analyze" to parse child logs
+    :setup:
+        1. Configure sssd to authenticate against directory server
+        2. Add user
+        3. Enable debug_level to 9 in the 'nss', 'pam' and domain section
+        4. Start SSSD
+    :steps:
+        1. Log in as user via ssh
+        2. Call sssctl analyze to check logs
+        3. Clear cache and restart SSSD
+        4. Log in as user via ssh with wrong password
+        5. Call sssctl analyze to check logs
+    :expectedresults:
+        1. Logged in successfully
+        2. Logs contain login related logs
+        3. Succesfully
+        4. Failed to login
+        5. Logs contain info about failed login
+    :customerscenario: True
+    """
+    provider.user("user1").add()
+    client.sssd.nss["debug_level"] = "9"
+    client.sssd.pam["debug_level"] = "9"
+    client.sssd.domain["debug_level"] = "9"
+    client.sssd.start()
+
+    client.ssh("user1", "Secret123").connect()
+
+    result = client.sssctl.analyze_request("show --pam --child 1")
+    assert result.rc == 0
+    assert "sss_child_krb5_trace_cb" in result.stdout
+    assert "user1@test" in result.stdout
+
+    client.sssd.stop()
+    client.sssd.clear(db=True, memcache=True, logs=True)
+    client.sssd.start()
+    time.sleep(5)
```

Comment:
why sleep here?

Reply:
I thought that it is needed because of clearing files. But as I am testing it, it seems that the sleep is not needed.

Reply:
Okay feel free to `Resolve conversation` once you test it and verify if it is needed or not.

Reply:
OK